core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala

```diff
@@ -153,15 +155,21 @@ class CollectorService(
 
         val (httpResponse, badRedirectResponses) =
           buildHttpResponse(event, params, headers.toList, redirect, pixelExpected, bounce, config.redirectMacro)
-        (httpResponse, badRedirectResponses ++ sinkResponses)
+
+        if (sinksHealthy) {
+          (httpResponse, badRedirectResponses ++ sinkResponses)
+        } else {
+          (HttpResponse(StatusCodes.ServiceUnavailable), badRedirectResponses ++ sinkResponses)
+        }
 
       case Left(error) =>
         val badRow = BadRow.GenericError(
           Processor(generated.BuildInfo.name, generated.BuildInfo.version),
           Failure.GenericFailure(Instant.now(), NonEmptyList.one(error.getMessage)),
           Payload.RawPayload(queryString.getOrElse(""))
         )
-        // We cannot return 400 to the JS tracker, as that will lead to the event getting eternally stuck in the client cache.
+
+        // No need to check the bad sink health. Even if it's unhealthy, we don't want the tracker to retry sending these events.
```

Comment:
Won't this potentially lead to an OOM if enough "bad" events get sent to the Collector and there is no ability to push these to the downstream targets?  If we do not want the tracker to retry them should they simply be dropped instead to avoid breaking the collector?

Reply:
I changed this to not trying to store bad rows in case of outage.